### PR TITLE
Updated examples, removed NewSoftSerial, small fixes

### PR DESCRIPTION
**Pull request by drak7**
Examples previously were set up for different boards, which worked but there was some duplication and it drove Travis crazy. They're now split into HardwareSerial and SoftwareSerial, and the appropriate boards can use whichever one they need. There are some ifdefs to keep Travis happy when it compiles for different platforms, including a stub that lets it compile a blank sketch when the board isn't supported.
I kept the shield_sdlog example even though it only uses SoftwareSerial because logging to an SD card is cool.

### CODE REVIEW

File: examples/GPS_HardwareSerial_EchoTest/GPS_HardwareSerial_EchoTest.ino [hunk 1]
```diff
@@ -41,3 +44,8 @@ void loop() {
     Serial.write(c);
   }
 }
+
+#else // Do nothing for boards without hardware serial
```
Comment (ladyada) [inline examples/GPS_HardwareSerial_EchoTest/GPS_HardwareSerial_EchoTest.ino]: i think we could print an #error but doesnt make sense to do nothing - they'll think its just not working :)

File: examples/GPS_HardwareSerial_LOCUS_DumpBasic/GPS_HardwareSerial_LOCUS_DumpBasic.ino [hunk 2]
```diff
@@ -64,3 +67,8 @@ void loop()                     // run over and over again
     Serial.write(c);
   }
 }//loop
+
+#else // Do nothing for boards without hardware serial
```
Comment (ladyada) [inline examples/GPS_HardwareSerial_LOCUS_DumpBasic/GPS_HardwareSerial_LOCUS_DumpBasic.ino]: ditto

File: examples/GPS_HardwareSerial_LOCUS_Erase/GPS_HardwareSerial_LOCUS_Erase.ino [hunk 3]
```diff
@@ -67,3 +70,7 @@ void loop()                     // run over and over again
   }
 }
 
+#else // Do nothing for boards without hardware serial
```
Comment (ladyada) [inline examples/GPS_HardwareSerial_LOCUS_Erase/GPS_HardwareSerial_LOCUS_Erase.ino]: ditto

File: examples/GPS_HardwareSerial_LOCUS_Start/GPS_HardwareSerial_LOCUS_Start.ino [hunk 4]
```diff
@@ -71,4 +74,7 @@ void loop()                     // run over and over again
 /******************************************************************/
 
 
-
+#else // Do nothing for boards without hardware serial
```
Comment (ladyada) [inline examples/GPS_HardwareSerial_LOCUS_Start/GPS_HardwareSerial_LOCUS_Start.ino]: ditto

File: examples/GPS_HardwareSerial_LOCUS_Status/GPS_HardwareSerial_LOCUS_Status.ino [hunk 5]
```diff
@@ -65,38 +68,43 @@ void loop()                     // run over and over again
...
       Serial.print((int)GPS.LOCUS_records); Serial.print(" Records, ");
-      Serial.print((int)GPS.LOCUS_percent); Serial.print("% Used "); 
-  
+      Serial.print((int)GPS.LOCUS_percent); Serial.print("% Used ");
+
     }//if (GPS.LOCUS_ReadStatus())
   }//if (millis() > updateTime)
 }//loop
+
+#else // Do nothing for boards without hardware serial
```
Comment (ladyada) [inline examples/GPS_HardwareSerial_LOCUS_Status/GPS_HardwareSerial_LOCUS_Status.ino]: ditto

File: examples/GPS_HardwareSerial_Parsing/GPS_HardwareSerial_Parsing.ino [hunk 6]
```diff
@@ -104,4 +107,9 @@ void loop() // run over and over again
       Serial.print("Satellites: "); Serial.println((int)GPS.satellites);
     }
   }
-}
\ No newline at end of file
+}
+
+#else // Do nothing for boards without hardware serial
```
Comment (ladyada) [inline examples/GPS_HardwareSerial_Parsing/GPS_HardwareSerial_Parsing.ino]: ditto

File: examples/GPS_SoftwareSerial_EchoTest/GPS_SoftwareSerial_EchoTest.ino [hunk 7]
```diff
@@ -74,3 +66,8 @@ void loop() {
     Serial.write(c);
   }
 }
+
+#else // Do nothing for other boards
```
Comment (ladyada) [inline examples/GPS_SoftwareSerial_EchoTest/GPS_SoftwareSerial_EchoTest.ino]: ditto

File: examples/GPS_SoftwareSerial_LOCUS_DumpBasic/GPS_SoftwareSerial_LOCUS_DumpBasic.ino [hunk 8]
```diff
@@ -58,17 +49,20 @@ void setup()
...
-      UDR0 = c;  
+      UDR0 = c;
 #else
       Serial.print(c);
 #endif
     }
   }
 }
 
+#else // Do nothing for other boards
```
Comment (ladyada) [inline examples/GPS_SoftwareSerial_LOCUS_DumpBasic/GPS_SoftwareSerial_LOCUS_DumpBasic.ino]: ditto

File: examples/GPS_SoftwareSerial_LOCUS_Erase/GPS_SoftwareSerial_LOCUS_Erase.ino [hunk 9]
```diff
@@ -5,60 +5,55 @@
...
 
 // Set GPSECHO to 'false' to turn off echoing the GPS data to the Serial console
 // Set to 'true' if you want to debug and listen to the raw GPS sentences
 #define GPSECHO  false
 
 // this keeps track of whether we're using the interrupt
 // off by default!
+#ifndef ESP8266 // Sadly not on ESP8266
 boolean usingInterrupt = false;
 void useInterrupt(boolean); // Func prototype keeps Arduino 0023 happy
```
Comment (ladyada) [inline examples/GPS_SoftwareSerial_LOCUS_Erase/GPS_SoftwareSerial_LOCUS_Erase.ino]: i think you can remove this line now?

File: examples/GPS_SoftwareSerial_LOCUS_Erase/GPS_SoftwareSerial_LOCUS_Erase.ino [hunk 10]
```diff
@@ -105,5 +101,9 @@ void useInterrupt(boolean v) {
     usingInterrupt = false;
   }
 }
+#endif // ESP8266
 
-
+#else // Do nothing for other boards
```
Comment (ladyada) [inline examples/GPS_SoftwareSerial_LOCUS_Erase/GPS_SoftwareSerial_LOCUS_Erase.ino]: ditto

File: examples/GPS_SoftwareSerial_LOCUS_Start/GPS_SoftwareSerial_LOCUS_Start.ino [hunk 11]
```diff
@@ -7,43 +7,37 @@
...
 
 // Set GPSECHO to 'false' to turn off echoing the GPS data to the Serial console
 // Set to 'true' if you want to debug and listen to the raw GPS sentences
 #define GPSECHO  true
 
 // this keeps track of whether we're using the interrupt
 // off by default!
+#ifndef ESP8266 // Sadly not on ESP8266
 boolean usingInterrupt = false;
 void useInterrupt(boolean); // Func prototype keeps Arduino 0023 happy
```
Comment (ladyada) [inline examples/GPS_SoftwareSerial_LOCUS_Start/GPS_SoftwareSerial_LOCUS_Start.ino]: try removing

File: examples/GPS_SoftwareSerial_LOCUS_Start/GPS_SoftwareSerial_LOCUS_Start.ino [hunk 12]
```diff
@@ -108,5 +106,9 @@ void useInterrupt(boolean v) {
     usingInterrupt = false;
   }
 }
+#endif // ESP8266
 
-
+#else // Do nothing for other boards
```
Comment (ladyada) [inline examples/GPS_SoftwareSerial_LOCUS_Start/GPS_SoftwareSerial_LOCUS_Start.ino]: ditto

File: examples/GPS_SoftwareSerial_LOCUS_Status/GPS_SoftwareSerial_LOCUS_Status.ino [hunk 13]
```diff
@@ -149,7 +144,9 @@ void useInterrupt(boolean v) {
     usingInterrupt = false;
   }
 }
+#endif // ESP8266
 
-
-
-
+#else // Do nothing for other boards
```
Comment (ladyada) [inline examples/GPS_SoftwareSerial_LOCUS_Status/GPS_SoftwareSerial_LOCUS_Status.ino]: ditto

File: examples/GPS_SoftwareSerial_Parsing/GPS_SoftwareSerial_Parsing.ino [hunk 14]
```diff
@@ -107,17 +98,22 @@ void loop()                     // run over and over again
...
+
       Serial.print("Speed (knots): "); Serial.println(GPS.speed);
       Serial.print("Angle: "); Serial.println(GPS.angle);
       Serial.print("Altitude: "); Serial.println(GPS.altitude);
       Serial.print("Satellites: "); Serial.println((int)GPS.satellites);
     }
   }
 }
+
+#else // Do nothing for other boards
```
Comment (ladyada) [inline examples/GPS_SoftwareSerial_Parsing/GPS_SoftwareSerial_Parsing.ino]: ditto

File: examples/shield_sdlog/shield_sdlog.ino [hunk 15]
```diff
@@ -206,6 +207,7 @@ void loop() {
   }
 }
 
-
-/* End code */
-
+#else // Do nothing for other boards
```
Comment (ladyada) [inline examples/shield_sdlog/shield_sdlog.ino]: ditto

File: examples/GPS_SoftwareSerial_LOCUS_Start/GPS_SoftwareSerial_LOCUS_Start.ino (hunk 11 above)
Comment (drak7) [reply examples/GPS_SoftwareSerial_LOCUS_Start/GPS_SoftwareSerial_LOCUS_Start.ino]: 8266 doesn't have TIMER0_COMPA_vect so the interrupt part fails when building for those.

Comment (ladyada) [reply examples/GPS_SoftwareSerial_LOCUS_Start/GPS_SoftwareSerial_LOCUS_Start.ino]: woops it marked the wrong line, or i was clumsy
i meant `void useInterrupt(boolean); // Func prototype keeps Arduino 0023 happy`

**Comment by ladyada** [comment]
reviewed! mostly please remove the 'do nothing' code cause otherwise people will run it and be confused when it compiles but does nothing :)
you can use test skip files if you want to not have that example checked
https://github.com/adafruit/travis-ci-arduino/blob/master/install.sh#L275
e.g. `.esp8266.test.skip`

**Comment by drak7** [comment]
Ah, didn't know about the skip files, I'll update everything.

**Comment by ladyada** [comment]
my fault - i forgot to tell you :D

**Comment by drak7** [comment]
No worries, that works better!

**Comment by ladyada** [comment]
whew!

**Pull request merged**